src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,362 @@
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                os = OSPlatform.Windows;
+                osName = "win";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                os = OSPlatform.Linux;
+                osName = "linux";
+            }
+            else
+            {
+                throw new ApplicationException("OS is not supported");
+            }
+
+            var downloadUrl = GetReleaseDownloadUrl(assets, osName);
+            var fileName = $"{osName}_{latest}.zip";
+            var directoryName = $"{fileName}temp";
+            var cts = new CancellationTokenSource();
+            var handler = new IoHandler(cts);
+
+            var progressWindow = new IoProgressWindow {
+                Title = $"Downloading {downloadUrl}",
+                Handler = handler
+            };
+            _ = progressWindow.ShowDialog(Program.Desktop.MainWindow);
+
+
+            // Download and unzip to a temp directory
+            var downloadSuccess = await HttpDownload(new Uri(downloadUrl).AbsoluteUri, fileName, handler, cts);
+            if (!downloadSuccess)
+            {
+                Logger.Info("Update download was cancelled");
+                return;
+            }
+
+
+            if (Directory.Exists(directoryName))
+                Directory.Delete(directoryName, true);
+
+            var progress = DownloadProgressPercentage / 100.0f;
+
+            Directory.CreateDirectory(directoryName);
+            using (ZipArchive archive = ZipFile.OpenRead(fileName))
+            {
+                foreach (ZipArchiveEntry entry in archive.Entries)
+                {
+                    await using Stream zipStream = entry.Open();
+
+                    var deltaProgress = 1 / (float)archive.Entries.Count * (100 - DownloadProgressPercentage);
+                    progress += deltaProgress;
+                    handler.UpdateProgress(deltaProgress, $"{float.Floor(DownloadProgressPercentage + progress)}%");
+
+                    var f = new FileStream($"{directoryName}/{entry.Name}", FileMode.CreateNew);
+                    await zipStream.CopyToAsync(f, cts.Token);
+                    f.Close();
+                }
+            }
+
+            var currentDirectory = Directory.GetCurrentDirectory();
+
+            // Wait 2 seconds for the process to fully exit before
+            // copying files from the temp directory to the root directory,
+            // then delete the temp directory and run the program again
+            var command =
+                os == OSPlatform.Linux
+                    ? $"""
+                           sleep 2 &&
+                           yes | cp -rf "{currentDirectory}\{directoryName}" &&
+                           rm "{currentDirectory}\{fileName}" &&
+                           rm -rf "{currentDirectory}\{directoryName}" &&
+                           ./Lumper.UI
```

Comment:
- copy doesn't have a destination
- `rm -rf` is scary, try moving files instead
- you can't rely on the current directory being where `./Lumper.UI` is
- forward slashes in the path